Automattic/woocommerce-payments
# Discoverability banner for Fraud & Risk tools

Fixes [#2785 on the server](https://github.com/Automattic/woocommerce-payments-server/issues/2785)

#### Changes proposed in this Pull Request



Adds Discoverability banner for Fraud & Risk tools to the WooCommerce Payments Overview screen







#### Testing instructions





* Checkout this branch.
* All tests should pass.
* Activate fraud and Risk tools feature flag option: `npm run wp option set wcpay_fraud_protection_settings_active 1`.
* Apply the patch included below.

#### Laptop Testing:
1. Navigate to the WCPay Overview page: `/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview`.
2. Inspect the banner. It should look like this:

![frt-banner-two-buttons](https://user-images.githubusercontent.com/60988591/225720197-e4241ac2-5099-4af2-8326-e2a4150385a5.png)

3. Click the 'Learn more' button. You should be directed to the WCPay settings screen. Once the components finish loading, you should see the screen smoothly scroll down to the Fraud Protection section.
4. Go back to the WCPay Overview page: `/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview`.
5. Click 'Remind me later'. The banner should disappear. 
6. The patch sets the `remindMeAt` timestamp to one minute in the future. So quickly click the Deposits link just below the Overview link in the left sidebar. Once the Deposits screen loads, click back to the Overview screen. The banner should not appear. Now click back to the Deposits screen and wait until the minute has fully elapsed. 
7. After one minute has passed, click the Overview link in the sidebar. The banner should appear.
8. Repeat steps 5 through 7, two more times.
9. Once you visit the Overview screen after clicking the 'Remind me later' button three times, the banner should appear, but this time it should include a 'Don't show me this again' button. The banner should look like this:

![frt-banner-three-buttons](https://user-images.githubusercontent.com/60988591/225721579-14a14ed6-e028-4b97-bcd5-0c974d756c06.png)

10. Click the 'Don't show me this again' button. The banner should disappear. Now click around to different admin screens and the Overview screen periodically for the next minute. You should not see the banner render on the Overview screen any more.

#### Mobile Testing:
1. Delete the banner's option from the database: `npm run wp option delete wcpay_frt_discover_banner_settings`.
2. Fire up Jurassic Tube (or Ngrok). JT: `npm run tube:start`.
3. Login to your WCPay test site via your smartphone.
4. Navigate to the WCPay Overview page: `/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview`. 
5. Inspect the banner. It should look something like this:
 
![frt-banner-two-buttons-mobile](https://user-images.githubusercontent.com/60988591/225724298-bf63e5bd-6894-4b76-af82-0c2fea125cd9.png)

6. Repeat steps 3 through 8 in the **Laptop Testing** section above. 
7. Once you visit the Overview screen after clicking the 'Remind me later' button three times, the banner should appear, but this time it should include a 'Don't show me this again' button. The banner should look something like this:

![frt-banner-three-buttons-mobile](https://user-images.githubusercontent.com/60988591/225725361-5c939b87-9732-41ce-a413-6b70d0aee295.png)

8. Click the 'Don't show me this again' button. The banner should disappear. Now click around to different admin screens and the Overview screen periodically for the next minute. You should not see the banner render on the Overview screen any more.

-------------------

<details>
  <summary><h3>Patch</h3></summary>

```patch
diff --git a/client/components/fraud-risk-tools-banner/index.js b/client/components/fraud-risk-tools-banner/index.js
index 3036275c..271a0d03 100644
--- a/client/components/fraud-risk-tools-banner/index.js
+++ b/client/components/fraud-risk-tools-banner/index.js
@@ -23,6 +23,8 @@ const FRTDiscoverabilityBanner = () => {
 		}
 	} );
 
+	console.log( settings );
+
 	const showBanner =
 		( null === settings.remindMeAt && ! settings.dontShowAgain ) ||
 		( Date.now() > settings.remindMeAt && ! settings.dontShowAgain )
@@ -30,13 +32,14 @@ const FRTDiscoverabilityBanner = () => {
 			: false;
 
 	const setReminder = () => {
-		const DAY_IN_MS = 24 * 60 * 60 * 1000;
+		// const DAY_IN_MS = 24 * 60 * 60 * 1000;
+		const MIN_IN_MS = 60 * 1000;
 		const nowTimestamp = Date.now();
 		setSettings( ( prevSettings ) => {
 			return {
 				...prevSettings,
 				remindMeCount: prevSettings.remindMeCount + 1,
-				remindMeAt: nowTimestamp + 3 * DAY_IN_MS,
+				remindMeAt: nowTimestamp + MIN_IN_MS,
 			};
 		} );
 	};

```

</details>

-------------------


**Post merge**
